### PR TITLE
[16.0][MIG] purchase_order_weight_volume: Migrate to version 16.0

### PR DESCRIPTION
**Pull request by tuantrantg**

### CODE REVIEW

**Review by leemannd — APPROVED** [review]

**Review by cyrilmanuel — APPROVED** [review]

File: purchase_order_weight_volume/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "name": "Purchase Order Weight and Volume",
+    "summary": """Display purchase order weight and volume""",
+    "version": "16.0.1.0.0",
+    "author": "Ilyas, Ooops404, Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/purchase-workflow",
+    "category": "Purchase Management",
+    "license": "AGPL-3",
```
Comment (simahawk) [inline purchase_order_weight_volume/__manifest__.py:11]: I see no license and no (c) in all the files... can we fix that?
Maybe to be done on 15 first and ported here. 
CC @ilyasProgrammer

File: purchase_order_weight_volume/models/purchase_order.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                        po.total_weight = sum(po.mapped("order_line.line_weight"))
+                        po.weight_uom_name = weight_uoms[0]
+
+            if po.company_id.display_order_volume_in_po:
+                volume_uoms = po.mapped("order_line.product_id.volume_uom_name")
+                if len(volume_uoms) > 0:
+                    same_volume_uom = all(el == volume_uoms[0] for el in volume_uoms)
+                    if same_volume_uom:
+                        po.total_volume = sum(po.mapped("order_line.line_volume"))
+                        po.volume_uom_name = volume_uoms[0]
```
Comment (simahawk) [inline purchase_order_weight_volume/models/purchase_order.py:65]: any chance to have some tests?

**Review by simahawk — CHANGES_REQUESTED** [review]

File: purchase_order_weight_volume/__manifest__.py (hunk 1 above)
Comment (ilyasProgrammer) [reply purchase_order_weight_volume/__manifest__.py:11]: @simahawk https://github.com/OCA/purchase-workflow/pull/1978

Comment (tuantrantg) [reply purchase_order_weight_volume/__manifest__.py:11]: @simahawk [#1978](https://github.com/OCA/purchase-workflow/pull/1978) is ported here.

File: purchase_order_weight_volume/models/purchase_order.py (hunk 2 above)
Comment (tuantrantg) [reply purchase_order_weight_volume/models/purchase_order.py:65]: @simahawk #1987 is taking care of the tests

Comment (tuantrantg) [reply purchase_order_weight_volume/models/purchase_order.py:65]: @simahawk tests included

**Comment by rafaelbn** [comment]
@simahawk I think you comments are attended could you please update your review? Thank you!

File: purchase_order_weight_volume/tests/test_purchase_order_weight_volume.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+from odoo.tests.common import TransactionCase
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class TestPurchaseOrderWeightVolume(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
```
Comment (simahawk) [inline purchase_order_weight_volume/tests/test_purchase_order_weight_volume.py:13]: One more thing please: set tracking_disable

**Comment by santostelmo** [comment]
@ping @simahawk 
Unit test added https://github.com/OCA/purchase-workflow/pull/1972/commits/adb44b436468a5088fbfede01c4db4bfc2676957

**Review by simahawk — APPROVED** [review]
pre-approving

File: purchase_order_weight_volume/tests/test_purchase_order_weight_volume.py (hunk 3 above)
Comment (tuantrantg) [reply purchase_order_weight_volume/tests/test_purchase_order_weight_volume.py:13]: hi @simahawk , `tracking_disable` is set, please review it again, thank you

**Comment by simahawk** [comment]
@tuantrantg pls make pre-commit happy

File: purchase_order_weight_volume/tests/test_purchase_order_weight_volume.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        cls.env = cls.env['base'].with_context(tracking_disable=True).env
```
Comment (simahawk) [inline purchase_order_weight_volume/tests/test_purchase_order_weight_volume.py]:
```suggestion
        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
```

Comment (tuantrantg) [reply purchase_order_weight_volume/tests/test_purchase_order_weight_volume.py]: Hi @simahawk PR updated

**Pull request merged**